persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -484,7 +485,18 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
         if (parameters.protocolVersion().isGreaterOrEqualTo(ProtocolVersion.V4))
           ClientWarn.instance.captureWarnings();
 
-        Single<QueryState> queryState = newQueryState();
+        Single<QueryState> queryState =
+            newQueryState()
+                .doOnEvent(
```

Comment:
I find calling this here very confusing, what about using `doOnSubscribe()` when firing up the request chain [here](https://github.com/stargate/stargate/pull/786/files#diff-0411e9be5edd1fdbfc193db9d75fc9c254d6030806bc90a27a19c9d45d0fb1e0R510)? That should work the same, and provide a more readable, closer to the "source", solution.

Reply:
Yes, I was thinking about `doOnSubscribe()`... will change.

The `here` link above does not seem to work for me, sorry. Could you just name the file and line number? @sbtourist ?

Reply:
Switched to `doOnSubscribe` and fixed the handling of warnings.

Reply:
> The here link above does not seem to work for me, sorry. Could you just name the file and line number?

I mean where `request.execute()` is called just below.

Reply:
Updated.